src/Products/ZopeVersionControl/nonversioned.py

```diff
@@ -163,3 +166,18 @@ def restoreNonVersionedData(self, data):
                 # a BTreeFolder2, which doesn't need or want the
                 # _objects attribute.
                 # XXX This is a hackish way to check for BTreeFolder2s.
+        # Yes, we're repeating ourselves:
+        if not hasattr(obj, '_tree'):
```

Comment:
I am missing context and knowledge why this needs to be done if we have no `_tree` :)

Reply:
That seems explained in the existing comment above it: when there is no `_tree` attribute, the object is a BTreeFolder which does not care about an `_objects` attribute, so we don't need to do anything.
I don't know enough to know if this is true, but I trust the original comment.